ksql-engine/src/main/java/io/confluent/ksql/ddl/commands/RegisterTopicCommand.java

```diff
@@ -81,6 +89,16 @@ private void enforceTopicProperties(final Map<String, Expression> properties) {
       throw new KsqlException("Topic format(format) should be set in WITH clause.");
     }
 
+    if (properties.containsKey(DdlConfig.DELIMITER_FORMAT_PROPERTY) && (
```

Comment:
It would be clearer if this block was extracted into a method, i..e, `validateFormatProperties` or similar.